packages/@aws-cdk/aws-codepipeline-actions/test/lambda/test.lambda-invoke-action.ts

```diff
@@ -0,0 +1,95 @@
+import { expect, haveResourceLike } from "@aws-cdk/assert";
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import lambda = require('@aws-cdk/aws-lambda');
+import { SecretValue, Stack, Token } from "@aws-cdk/cdk";
+import { Test } from 'nodeunit';
+import cpactions = require('../../lib');
+
+// tslint:disable:object-literal-key-quotes
+
+export = {
+  'Lambda invoke Action': {
+    'properly serializes the object passed in userParameters'(test: Test) {
+      const stack = stackIncludingLambdaInvokeCodePipeline({
+        key: 1234,
+      });
+
+      expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
+        "Stages": [
+          {},
+          {
+            "Actions": [
+              {
+                "Configuration": {
+                  "UserParameters": '{"key":1234}',
+                },
+              },
+            ],
+          },
+        ],
+      }));
+
+      test.done();
+    },
+
+    'properly resolves any Tokens passed in userParameters'(test: Test) {
+      const stack = stackIncludingLambdaInvokeCodePipeline({
+        key: new Token(() => null),
```

Comment:
Why `null`? Might make more sense to use some realistic example like `Aws.region` or something...

Reply:
Changed to something more interesting.